client/ayon_core/pipeline/workfile/workfile_template_builder.py

```diff
@@ -506,55 +506,52 @@ def build_template(
             keep_placeholders (bool): Add flag to placeholder data for
                 hosts to decide if they want to remove
                 placeholder after it is used.
-            create_first_version (bool): create first version of a workfile
-            workfile_creation_enabled (bool): If True, it might create
-                                              first version but ignore
-                                              process if version is created
+            create_first_version (bool): Create first version of a workfile.
+                 When set to True, this option initiates the saving of the
+                 workfile for an initial version. It will skip saving if
+                 a version already exists.
+            workfile_creation_enabled (bool): Whether we are creating a new
+                workfile. If False, we assume we just want to build the
+                template in our current scene.
 
         """
-        if any(
-            value is None
-            for value in [
-                template_path,
-                keep_placeholders,
-                create_first_version,
-            ]
+
+        # Get default values if not provided
+        if (
+                template_path is None
+                or keep_placeholders is None
+                or create_first_version is None
         ):
-            template_preset = self.get_template_preset()
-            if template_path is None:
-                template_path = template_preset["path"]
+            preset = self.get_template_preset()
+            template_path: str = template_path or preset["path"]
             if keep_placeholders is None:
-                keep_placeholders = template_preset["keep_placeholder"]
+                keep_placeholders: bool = preset["keep_placeholder"]
             if create_first_version is None:
-                create_first_version = template_preset["create_first_version"]
-
-        # check if first version is created
-        created_version_workfile = False
-        if create_first_version:
-            created_version_workfile = self.create_first_workfile_version()
+                create_first_version: bool = preset["create_first_version"]
 
-        # if first version is created, import template
-        # and populate placeholders
-        if (
-            create_first_version
-            and workfile_creation_enabled
-            and created_version_workfile
-        ):
+        # Build the template if current workfile is a new unsaved file
+        # (that's detected by checking if it returns any current filepath)
+        if not self.host.get_current_workfile():
```

Comment:
Actually - this check is incorrect. We _should_ be allowed to build the template even in an existing scene - import an additional template and build it. I guess I did make a fault in my thinking. The `build_template` is NOT only triggered on launch of new workfiles, but it is ALSO a separate function in some menus...